src/AnalyzerPowerPack/Core/Design/CA1052DiagnosticAnalyzer.cs

```diff
@@ -100,13 +104,18 @@ internal static class CA1052Extensions
         /// "qualifying member" (<see cref="IsQualifyingMember(ISymbol)"/>) and no
         /// "disqualifying members" (<see cref="IsDisqualifyingMember(ISymbol)"/>).
         /// </remarks>
-        internal static bool IsStaticHolderType(this INamedTypeSymbol symbol)
+        internal static bool IsStaticHolderType(this INamedTypeSymbol symbol, Compilation compilation)
         {
             if (symbol.TypeKind != TypeKind.Class)
             {
                 return false;
             }
 
+            if (symbol.BaseType != compilation.GetTypeByMetadataName("System.Object"))
```

Comment:
Yes, there is a base class in this case: ExtendedErrorTypeSymbol with SpecialType==None, and null check still not needed. Added test cases to cover this.